src/pfe/portal/modules/projectInitializer.js

```diff
@@ -189,10 +189,17 @@ const createEnvForContainerCreation = (user, projectName, gitInfo) => {
   env.push(`CW_EXTENSIONS=${JSON.stringify(cwExtensions)}`);
 
   if (projectName && gitInfo.repo) {
-    env.push(
-      `PROJ_NAME=${projectName}`,
-      `GIT_REPO=${gitInfo.repo}`,
-    );
+    if (gitInfo.repo.endsWith("tar.gz")) {
+      env.push(
+        `PROJ_NAME=${projectName}`,
+        `TAR_FILE=${gitInfo.repo}`,
+      );
+    } else {
+      env.push(
+        `PROJ_NAME=${projectName}`,
+        `GIT_REPO=${gitInfo.repo}`,
+      );
+    }
```

Comment:
The `gitInfo.branch` check can go into this `else` block right?  Since it would be applicable only for non-tarball scenario